src/controller/JCMCommissioner.cpp

```diff
@@ -0,0 +1,438 @@
+{
+    auto attributeCache = info.attributes;
+
+    CHIP_ERROR err = attributeCache->ForEachAttribute(OperationalCredentials::Id, [this, &attributeCache](const ConcreteAttributePath & path) {
+        using namespace chip::app::Clusters::OperationalCredentials::Attributes;
+
+        switch (path.mAttributeId)
+        {
+            case Fabrics::Id: {
+                Fabrics::TypeInfo::DecodableType fabrics;
+                ReturnErrorOnFailure(attributeCache->Get<Fabrics::TypeInfo>(path, fabrics));
+
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    if (fabricDescriptor.fabricIndex != kUndefinedFabricIndex)
+                    {
+                        if (fabricDescriptor.rootPublicKey.size() != Crypto::kP256_PublicKey_Length)
+                        {
+                            ChipLogError(Controller, "JCM: DeviceCommissioner::ParseJFAdministratorInfo - fabric root key size mismatch");
+                            return CHIP_ERROR_KEY_NOT_FOUND;
+                        }
+
+                        mInfo.rootKeySpan = fabricDescriptor.rootPublicKey;
+                        mInfo.adminVendorId = fabricDescriptor.vendorID;
+                        mInfo.adminFabricId = fabricDescriptor.fabricID;
+
+                        if (fabricDescriptor.VIDVerificationStatement.HasValue())
+                        {
+                            ChipLogError(Controller, "JCM: Per-home RCAC are not supported by JF for now!");
+                            return CHIP_ERROR_CANCELLED;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator Fabric Table");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
+            }
+            case NOCs::Id: {
+                NOCs::TypeInfo::DecodableType nocs;
+                ReturnErrorOnFailure(attributeCache->Get<NOCs::TypeInfo>(path, nocs));
+
+                auto iter = nocs.begin();
+                while (iter.Next())
+                {
+                    auto & nocStruct = iter.GetValue();
+
+                    if (nocStruct.fabricIndex == mInfo.adminFabricIndex)
+                    {
+                        if (!mInfo.adminNOC.Alloc(nocStruct.noc.size()))
+                        {
+                            ChipLogError(Controller, "JCM: DeviceCommissioner::ParseJFAdministratorInfo - cannot allocate memory for admin noc");
+                            return CHIP_ERROR_NO_MEMORY;
+                        }
+
+                        memcpy(mInfo.adminNOC.Get(), nocStruct.noc.data(), nocStruct.noc.size());
+
+                        if (!nocStruct.icac.IsNull())
+                        {
+                            auto icac = nocStruct.icac.Value();
+                            if (!mInfo.adminICAC.Alloc(icac.size()))
```

Comment:
In https://github.com/project-chip/connectedhomeip/pull/38749 all this memory management was factored out into a separate function so it's not copy/pasted....

Reply:
AllocateMemoryAndCopySpan was private to the AutoCommissioner so I moved it out to a new AllocatorUtils.[h,cpp] module in the  lib/support package for general re-use by both the AutoCommissioner, JCMCommissioner and other modules in the future.

Reply:
@bzbarsky-apple Moved into the ScopedBuffer.h.  See both CopyFromSpan and GetSpan methods.